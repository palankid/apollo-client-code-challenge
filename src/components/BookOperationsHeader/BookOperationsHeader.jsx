import React from 'react';
import { string } from 'prop-types';
import { BookOutlined } from '@ant-design/icons';

const BookOperationsHeader = ({ titleFragment }) => {
    return (
        <header className="book-operations-header">
            <BookOutlined
                className="book-operations-header__icon"
                size={50}
            />
            <span className="book-operations-header__title">
                Books / { titleFragment }
            </span>
        </header>
    );
}

BookOperationsHeader.propTypes = {
    titleFragment: string.isRequired
}

export default BookOperationsHeader;
