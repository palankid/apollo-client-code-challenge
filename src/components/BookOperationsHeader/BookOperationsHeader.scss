.book-operations-header {
    display: flex;
    background-color: #FFFFFF;
    padding: 20px 0;
    align-items: center;

    &__icon {
        font-size: 30px;
    }

    &__title {
        font-size: 18px;
        font-weight: bold;
        margin-left: 10px;
    }
}
