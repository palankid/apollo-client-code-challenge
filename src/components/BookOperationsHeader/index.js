import BookOperationsHeader from './BookOperationsHeader';
import './BookOperationsHeader.scss';

export default BookOperationsHeader;
