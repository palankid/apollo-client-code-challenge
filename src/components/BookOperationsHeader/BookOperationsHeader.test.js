import React from 'react';
import { shallow } from 'enzyme';

import BookOperationsHeader from './BookOperationsHeader';

describe('BookOperationsHeader', () => {
    it('should test that component is rendered and not changed unintentionally', () => {
        const wrapper = shallow(
            <BookOperationsHeader
                titleFragment="Test"
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
