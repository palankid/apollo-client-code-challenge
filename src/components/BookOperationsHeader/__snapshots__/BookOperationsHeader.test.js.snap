// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookOperationsHeader should test that component is rendered and not changed unintentionally 1`] = `
<header
  className="book-operations-header"
>
  <ForwardRef(BookOutlined)
    className="book-operations-header__icon"
    size={50}
  />
  <span
    className="book-operations-header__title"
  >
    Books / 
    Test
  </span>
</header>
`;
